src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * The parser for the duke program.
+ * Processes user string input.
+ */
+public class Parser {
+
+    /**
+     * Parses a user's input, and calls the necessary functions.
+     *
+     * @param input The user's input.
+     * @param tasks The TaskList being used.
+     * @param storage The storage object being used.
+     */
+    public static String parse(String input, TaskList tasks, Storage storage) {
```

Comment:
Should the `parse` method do only parsing command instead of calling the command?